.github/workflows/test.yml

```diff
@@ -340,70 +102,120 @@ jobs:
-              shell: bash -l {0}
-              working-directory: ${{ matrix.example_directories }}
-              run: AZLE_PROPTEST_NUM_RUNS=5 AZLE_PROPTEST_VERBOSE=true npm test
-            - if: ${{ needs.release-candidate-deploy.outputs.should_run_tests && contains(github.head_ref, 'release--') && !(github.ref == 'refs/heads/main' && contains(github.event.head_commit.message, 'Merge pull request') && contains(github.event.head_commit.message, 'demergent-labs/release--')) }}
-              shell: bash -l {0}
-              working-directory: ${{ matrix.example_directories }}
-              run: AZLE_PROPTEST_NUM_RUNS=10 AZLE_PROPTEST_VERBOSE=true npm test
-            - if: ${{ needs.release-candidate-deploy.outputs.should_run_tests && (github.ref == 'refs/heads/main' && contains(github.event.head_commit.message, 'Merge pull request') && contains(github.event.head_commit.message, 'demergent-labs/release--')) }}
+
+            - name: Run Test
+              run: |
+                  BRANCH_NAME="${{ github.head_ref }}"
+                  MAIN_BRANCH='refs/heads/main'
+                  COMMIT_MESSAGE="${{ github.event.head_commit.message }}"
+                  RELEASE_TAG='release--'
+                  RELEASE_PR_MERGE_MSG='Merge pull request'
+                  RELEASE_REPO_MSG='demergent-labs/release--'
+
+                  IS_FEATURE_PR=true
+                  IS_RELEASE_PR=false
+                  IS_MERGE_TO_MAIN_FROM_RELEASE=false
+
+                  # Check if it's a feature pull request
+                  if [[ $BRANCH_NAME != *"$RELEASE_TAG"* && ${{ github.ref }} != "$MAIN_BRANCH" && !($COMMIT_MESSAGE == *"$RELEASE_PR_MERGE_MSG"* && $COMMIT_MESSAGE == *"$RELEASE_REPO_MSG"*) ]]; then
+                      IS_FEATURE_PR=true
+                  else
+                      IS_FEATURE_PR=false
+                  fi
+
+                  # Check if it's a release candidate pull request
+                  if [[ $BRANCH_NAME == *"$RELEASE_TAG"* && ${{ github.ref }} != "$MAIN_BRANCH" && !($COMMIT_MESSAGE == *"$RELEASE_PR_MERGE_MSG"* && $COMMIT_MESSAGE == *"$RELEASE_REPO_MSG"*) ]]; then
+                      IS_RELEASE_PR=true
+                  else
+                      IS_RELEASE_PR=false
+                  fi
+
+                  # Check if it's a merge from a release candidate into main
+                  if [[ ${{ github.ref }} == "$MAIN_BRANCH" && $COMMIT_MESSAGE == *"$RELEASE_PR_MERGE_MSG"* && $COMMIT_MESSAGE == *"$RELEASE_REPO_MSG"* ]]; then
+                      IS_MERGE_TO_MAIN_FROM_RELEASE=true
+                  else
+                      IS_MERGE_TO_MAIN_FROM_RELEASE=false
+                  fi
+
+                  # Determine the number of runs based on the conditions
```

Comment:
Is this the best place for the variables to live?

Reply:
I think so, I consolidated one into IS_MAIN_BRANCH and moved it up with the others because it could just in case, maybe it's not too much future proofing... and the I obliterated as many of them as I could. I think the result is more declarative.